core/src/main/java/me/aurium/beetle/branch/block/BlockList.java

```diff
@@ -0,0 +1,13 @@
+package me.aurium.beetle.branch.block;
+
+import java.util.List;
+
+public interface BlockList extends List<Block>{
```

Comment:
I suggest composition instead of inheritance, e.g.
```suggestion
public interface BlockList {

  List<Block> getBackingList();
  
  void addFirst(Block block);
  void addLast(Block block);
  // ...
```
Doing so would also mean you don't have to implement List

Reply:
I'd usually prefer composition except in this case.